vllm/worker/model_runner.py

```diff
@@ -1426,12 +1426,6 @@ def capture_model(self, kv_caches: List[List[torch.Tensor]]) -> None:
                 dtype=self.model_config.dtype,
                 device=self.device)
 
-        # Prepare buffer for outputs. These will be reused for all batch sizes.
-        # It will be filled after the first graph capture.
-        hidden_or_intermediate_states: List[Optional[torch.Tensor]] = [
-            None
```

Comment:
this functionality is actually broken in the main branch. I designed this code in https://github.com/vllm-project/vllm/pull/5074 , and the key idea is to reuse the output buffer manually. However, the optimization is accidentally deleted in https://github.com/vllm-project/vllm/pull/4628/files#diff-d3df23c3e3bcfe97ee8507061c6de54f0eff23a8c75d7f5999062c42245290f8R993 .

nevertheless, with this pr, we don't need to manually reuse tensors like this.